raiden/channel.py

```diff
@@ -543,6 +543,32 @@ def __init__(self, our_state, partner_state, external_state,
         self.on_task_completed_callbacks = list()  # XXX naming
 
     @property
+    def status(self):
+        status = None
+        if self.isopen:
+            status = 'open'
+            return status
+        else:
+            if self.external_state.settled_block != 0:
+                status = 'settled'
+                if self.external_state.closed_block != 0:
+                    status = 'closed'
+                return status
+        raise Exception('invalid state')
```

Comment:
So closed_block also is != 0 when the channel is settled.
Ok, your checking this with the order of the elif cases, right. Yes, that's smoother, but we should add a comment that the order matters.